ci: GitHub Actions update to Visual Studio 16.4.0

On [2019-12-09](https://github.blog/changelog/2019-12-19-github-actions-updates-to-the-virtual-environments/) GitHub Actions virtual environments has been upgraded to Visual Studio 16.4.0.

This causes errors:
```
     6>LINK : fatal error C1047: The object or library file 'C:\vcpkg\installed\x64-windows-static\lib\libzmq-mt-s-4_3_3.lib' was created with an older compiler than other objects; rebuild old objects and libraries [D:\a\bitcoin\bitcoin\build_msvc\bitcoind\bitcoind.vcxproj]
     6>LINK : fatal error LNK1257: code generation failed [D:\a\bitcoin\bitcoin\build_msvc\bitcoind\bitcoind.vcxproj]
```

Updating Qt binaries (#17736) and removing `zeromq` package from `build_msvc/vcpkg-packages.txt` did not help.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -7,8 +7,8 @@ jobs:
     runs-on: windows-latest
     env:
         PYTHONUTF8: 1
-        QT_DOWNLOAD_URL: 'https://github.com/sipsorcery/qt_win_binary/releases/download/v1.4/Qt5.9.8_x64_static_vs2019.zip'
-        QT_DOWNLOAD_HASH: 'f285cbb02bec3b3f3cc2621e3fa7d5edf0d6a66fa30c57859e583acda954ea80'
+        QT_DOWNLOAD_URL: 'https://github.com/sipsorcery/qt_win_binary/releases/download/v1.6/Qt5.9.8_x64_static_vs2019.zip'
+        QT_DOWNLOAD_HASH: '9a8c6eb20967873785057fdcd329a657c7f922b0af08c5fde105cc597dd37e21'
         QT_LOCAL_PATH: 'C:\Qt5.9.8_x64_static_vs2019'
         VCPKG_INSTALL_PATH: "$env:VCPKG_INSTALLATION_ROOT/installed"
         PLATFORM: x64

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -19,11 +19,17 @@ jobs:
       with:
         python-version: '3.7'  # Needed for PEP 540
 
+    - name: Setup MSBuild.exe
+      uses: warrenbuckley/Setup-MSBuild@v1
+
+    - name: Check MSBuild.exe
+      run: MSBuild.exe -version | Out-File -FilePath $env:GITHUB_WORKSPACE\MSBuild_version
+
     - uses: actions/cache@v1
       id: vcpkgcache
       with:
        path: C:/vcpkg/installed
-       key: ${{ runner.os }}-vcpkg
+       key: ${{ runner.os }}-vcpkg-${{ hashFiles('MSBuild_version') }}
 
     - name: Update vcpkg and install packages
       if: steps.vcpkgcache.outputs.cache-hit != 'true'
@@ -54,8 +60,6 @@ jobs:
         }
     - name: Generate project files
       run: python build_msvc\msvc-autogen.py
-    - name: Setup MSBuild.exe
-      uses: warrenbuckley/Setup-MSBuild@v1
     - name: vcpkg integration
       run: C:/vcpkg/vcpkg.exe integrate install
     - name: Build

